src/Symfony/Component/PropertyInfo/PropertyWriteInfo.php

```diff
@@ -0,0 +1,133 @@
+    public const TYPE_ADDER_AND_REMOVER = 'adder_and_remover';
+    public const TYPE_CONSTRUCTOR = 'constructor';
+
+    public const VISIBILITY_PUBLIC = 'public';
+    public const VISIBILITY_PROTECTED = 'protected';
+    public const VISIBILITY_PRIVATE = 'private';
+
+    private $type;
+    private $name;
+    private $visibility;
+    private $static;
+    private $adderInfo;
+    private $removerInfo;
+
+    private function __construct()
+    {
+    }
+
+    public function getType(): string
+    {
+        return $this->type;
+    }
+
+    public function getName(): string
+    {
+        if (null === $this->name) {
+            throw new \LogicException("Calling getName() when having a mutator of type {$this->type} is not tolerated");
+        }
+
+        return $this->name;
+    }
+
+    public function getAdderInfo(): self
+    {
+        if (null === $this->adderInfo) {
+            throw new \LogicException("Calling getAdderInfo() when having a mutator of type {$this->type} is not tolerated");
+        }
+
+        return $this->adderInfo;
+    }
+
+    public function getRemoverInfo(): self
+    {
+        if (null === $this->removerInfo) {
+            throw new \LogicException("Calling getRemoverInfo() when having a mutator of type {$this->type} is not tolerated");
+        }
+
+        return $this->removerInfo;
+    }
+
+    public function getVisibility(): string
+    {
+        if (null === $this->visibility) {
+            throw new \LogicException("Calling getVisibility() when having a mutator of type {$this->type} is not tolerated");
+        }
+
+        return $this->visibility;
+    }
+
+    public function isStatic(): bool
+    {
+        if (null === $this->static) {
+            throw new \LogicException("Calling isStatic() when having a mutator of type {$this->type} is not tolerated");
+        }
+
+        return $this->static;
+    }
+
+    public static function forSetter(string $methodName, string $visibility, bool $static): self
+    {
+        $mutator = new self();
+        $mutator->type = self::TYPE_SETTER;
+        $mutator->name = $methodName;
+        $mutator->visibility = $visibility;
+        $mutator->static = $static;
+
+        return $mutator;
+    }
+
+    public static function forProperty(string $propertyName, string $visibility, bool $static): self
+    {
+        $mutator = new self();
+        $mutator->type = self::TYPE_PROPERTY;
+        $mutator->name = $propertyName;
+        $mutator->visibility = $visibility;
+        $mutator->static = $static;
+
+        return $mutator;
+    }
+
+    public static function forAdderAndRemover(self $adder, self $remover): self
+    {
+        $mutator = new self();
+        $mutator->type = self::TYPE_ADDER_AND_REMOVER;
+        $mutator->adderInfo = $adder;
+        $mutator->removerInfo = $remover;
+
+        return $mutator;
+    }
+
+    public static function forConstructor(string $propertyName): self
```

Comment:
Should we have a visibility here as well? Constructors could be private.

Reply:
I'm hesitating for that, constructor can be private / protected or public, but having the visiblity info on the property makes little sense (since the visiblity is on the method, not the property).

Can we keep this like that for the moment ? I admit the constructor thing is not the best ATM but would like to change signature and information when it will be used (it's only for the #30248 PR ATM) ?

Reply:
:+1: